bundles/framework/maplegend/Flyout.js

```diff
@@ -100,175 +105,47 @@ Oskari.clazz.define('Oskari.mapframework.bundle.maplegend.Flyout',
         refresh: function () {
             this._populateLayerList();
         },
+        showMetadataFlyout: function (event, uuid) {
+            event.stopPropagation();
+            this.sandbox.postRequestByName('catalogue.ShowMetadataRequest', [{
+                uuid: uuid
+            }]);
+        },
         /**
          * @method _populateLayerList
          * @private
          * @param {Object} layerListContainer reference to jQuery object representing the layerlist placeholder
          * Renders legend images as an accordion for the selected layers.
          */
         _populateLayerList: function () {
-            var me = this;
-            var cel = jQuery(this.container);
-            cel.empty();
-
-            var accordion = Oskari.clazz.create('Oskari.userinterface.component.Accordion');
-            accordion.insertTo(cel);
-
-            var sandbox = this.instance.getSandbox();
-
             // populate selected layer list
-            var layers = sandbox.findAllSelectedMapLayers().slice(0),
-                n,
-                layer,
-                layerContainer,
-                accordionPanel;
-
-            for (n = layers.length - 1; n >= 0; n -= 1) {
-                layer = layers[n];
-                layerContainer = this._createLayerContainer(layer);
+            const layers = this.sandbox.findAllSelectedMapLayers().slice(0);
+            const titles = [];
 
-                const layerTitle = jQuery('<div class="maplegend-layer-title">' + Oskari.util.sanitize(layer.getName()) + '</div>');
+            for (const layer of layers) {
```

Comment:
You could use map to make it a bit easier to read and you could take of that slice() for layers as we don't pass it past this method:
```suggestion
const showMetadata = (event) => this.showMetadataFlyout(event, uuid);
const titles = layers
    .filter(layer => 
        typeof layer.getLegendImage == 'function' && !!layer.getLegendImage())
    .map(layer => {
        const uuid = layer.getMetadataIdentifier();
        return {
            title: layer.getName(),
            uuid: uuid,
            legendImageURL: layer.getLegendImage(),
            loadError: false,
            showMetadataCallback: uuid ? showMetadata : null;
        };
    });
```
One change above is that `showMetadataCallback` is given if there's an `uuid`. Your code uses `layerLegendImage` and I'm not sure why would that be used as uuid is used to show metadata, not the legend.

Reply:
Maybe we should also rename titles -> legends as the variable?

Reply:
Renamed.